Doc/library/sqlite3.rst

```diff
@@ -47,85 +47,170 @@ This document includes four main sections:
       PEP written by Marc-André Lemburg.
 
 
+.. We use the following practises for SQL code:
+   - UPPERCASE for keywords
+   - snake_case for schema
+   - single quotes for string literals
+   - singular for table names
+   - if needed, use double quotes for table and column names
+
 .. _sqlite3-tutorial:
 
 Tutorial
 --------
 
-To use the module, start by creating a :class:`Connection` object that
-represents the database.  Here the data will be stored in the
-:file:`example.db` file::
+In this tutorial, you will learn the basics of the :mod:`!sqlite3` API
+by creating a database of Monty Python movies.
+It assumes a fundamental understanding of database concepts,
+including `cursors`_ and `transactions`_.
+
+First, you'll need to create a new database to hold the movie data
+and open a database connection to allow :mod:`!sqlite3` to work with it.
+Call the :func:`sqlite3.connect` function to
+to create a connection to a database :file:`tutorial.db`
+in the current working directory,
+implicitly creating it if it does not exist::
 
    import sqlite3
-   con = sqlite3.connect('example.db')
+   con = sqlite3.connect("tutorial.db")
 
-The special path name ``:memory:`` can be provided to create a temporary
-database in RAM.
+The returned :class:`Connection` object ``con``
+represents the connection to the database.
 
-Once a :class:`Connection` has been established, create a :class:`Cursor` object
-and call its :meth:`~Cursor.execute` method to perform SQL commands::
+In order to execute SQL statements and fetch results from SQL queries,
+you'll use a database cursor.
+Call :meth:`con.cursor() <Connection.cursor>` to create the :class:`Cursor`::
 
    cur = con.cursor()
 
-   # Create table
-   cur.execute('''CREATE TABLE stocks
-                  (date text, trans text, symbol text, qty real, price real)''')
-
-   # Insert a row of data
-   cur.execute("INSERT INTO stocks VALUES ('2006-01-05','BUY','RHAT',100,35.14)")
+Now you'll create a database table ``movie`` with columns for title,
```

Comment:
```suggestion
Now you can create a database table ``movie`` with columns for title,
```
You don't need to use just use "need" instead of "will" (as suggested above) -- you can also use "can", which still sounds less demanding.

Reply:
I'm not sure about this—[per Diataxis](https://diataxis.fr/tutorials/), the purpose of tutorials is to instruct the reader in the steps they "need" to complete to accomplish a basic goal with the module, not simply letting them know what they "can" do. And per [The language of tutorials](https://diataxis.fr/tutorials/#the-language-of-tutorials), the examples given all use clear, direct language telling the reader what _to_ do:

> First, do x. Now, do y. Now that you have done y, do z.

Reply:
I'm not sure how literally that example should be taken.  ISTM the point of the example was to show a clear and not ambiguous sequence of steps, rather than the words and tenses used.  Maybe @evildmp can confirm? (I'd also appreciate a look over the whole PR once we are done tweaking)

Reply:
Fair, though I wasn't just looking at that example—it was based off (at least my interpretation of) a number of other statements there, e.g. [emphasis original]

> The only responsibility of the pupil is to be attentive and to follow the teacher’s directions as closely as they can.

> *Your guidance needs to remain focused on what’s required to reach the conclusion*

On the other hand, it also states

> It helps to maintain a friendly tone,

I definitely appreciate the value in such, and we don't want to sound overly harsh—but we also don't want to be too equivocal or hesitant that clarity, directness or crispness is affected.

Reply:
Yes, it's about the clarity of sequence, and emphasising the specific actions that need to be taken.

The verbs do make a difference though, but I wouldn't say it's the difference between right and wrong. 

### _Can_

I think I'd only use _can_:

* if we've been working towards something that we've had to defer until this moment (_Now that we have initialised the database, we can create the table we need_)
* if I want to signal that we're doing something that's not strictly required (e.g. good practice: _We can shut down the database connection to save resources, so..._. Or maybe to explore: (_You can terminate and reopen the session at any time: [some command]_).

### _We_ vs _you_

Different projects have different styles. Still, my own preference is generally something like:

> Now we need to initialise the database, so run the appropriate command in the terminal:
>
> `some command whatever`

I like this because it puts the tutor and the learner together in the same situation, as though we were sitting over the same piece of work. But, it's the learner who is performing the actions. Our need, their activity; it's how we naturally speak to someone when we are teaching them a practical skill. 

Other reasons I like the first person plural:

* I wouldn't use it in a how-to guide (I would just say: _Initialise the database..._), so it become another way of maintaining a distinction between the forms of a tutorial and those of a how-to guide.
* I think it helps us as authors to see the situation the user is in, because we have placed ourselves in that situation too.

Reply:
> I like this because it puts the tutor and the learner together in the same situation, as though we were sitting over the same piece of work. But, it's the learner who is performing the actions. Our need, their activity; it's how we naturally speak to someone when we are teaching them a practical skill.

That is very insightful and connects well; I'd appreciate having at least some kind of guidance like that in Diataxis itself, as I went looking and didn't find any. I would "naturally" default to the same in that context myself, because I want the reader to feel as if we have a shared objective and we're in this together, but they are the one actually doing the work (which is typically how I frame it when I'm doing 1 on 1 tutoring). However, I couldn't find any clear guidance in Diataxis, and when I went looking elsewhere, some other sources strongly discouraged "we" for not centering the reader.

Reply:
Very insightful indeed. I believe we should consider using first person plural again before landing this, for the reasons provided.